inference-engine/src/transformations/src/transformations/op_conversions/gather_normalize_negative_indices.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/gather_normalize_negative_indices.hpp"
+
+#include <memory>
+
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::GatherNegativeConstIndicesNormalize, "GatherNegativeConstIndicesNormalize", 0);
+
+ngraph::pass::GatherNegativeConstIndicesNormalize::GatherNegativeConstIndicesNormalize() {
+    MATCHER_SCOPE(GatherNegativeConstIndicesNormalize);
+    auto data_input = ngraph::pattern::any_input(pattern::has_static_shape());
+    auto axis_input = ngraph::pattern::any_input();
+    auto const_indices_input = ngraph::pattern::wrap_type<ngraph::opset7::Constant>();
+    auto gather_node = std::make_shared<ngraph::opset7::Gather>(data_input, const_indices_input, axis_input);
```

Comment:
why is it implemented so complicated? Why don't you code as:
`
auto gather = ngraph::pattern::wrap_type<opset7::Gather>();
ngraph::matcher_pass_callback callback = ...
auto m = std::make_shared<ngraph::pattern::Matcher>(gather, matcher_name);
register_matcher(m, callback);
`

Reply:
I think it is better to give detailed description of gather inputs in pattern to avoid many additional checks in transformation which will make transformation code less understandable.